src/server/dfly_main.cc

```diff
@@ -299,6 +300,21 @@ string NormalizePaths(std::string_view path) {
   return string(path);
 }
 
+bool StoiOctalBase(string str, int* out) {
+  size_t pos;
+  try {
+    *out = std::stoi(str, &pos, 8);
+  } catch (const std::invalid_argument&) {
+    return false;
+  } catch (const std::out_of_range&) {
+    return false;
+  }
+  if (pos != str.size()) {
+    return false;
+  }
+  return true;
+}
```

Comment:
Oh, I didn't realize we'd have to use exceptions :(
Since you want exactly the format of `0XYZ`, maybe just write a custom function?
And also - maybe return a `std::optional<int>` instead of an out param?

Reply:
Please use safe_strtou32_base from absl. (yeah, it's internal, but lets still use it).